Tests broken due to https://github.com/DiamondLightSource/dodal/pull/544

https://github.com/DiamondLightSource/dodal/pull/544 has broken our tests, we should fix them

After fixing for the move of `beamline_utils` there are additional failures:
```
FAILED tests/unit_tests/experiment_plans/test_optimise_attenuation_plan.py::test_arm_devices_runs_correct_functions - AssertionError: Expected 'mock' to have been called once. Called 0 times.
FAILED tests/unit_tests/experiment_plans/test_pin_tip_centring.py::test_given_no_tip_found_but_will_be_found_when_get_tip_into_view_then_smargon_moved_positive_and_tip_returned - AssertionError: assert 0 == 0.5
FAILED tests/unit_tests/experiment_plans/test_pin_tip_centring.py::test_given_tip_at_zero_but_will_be_found_when_get_tip_into_view_then_smargon_moved_negative_and_tip_returned - AssertionError: assert 0 == -0.5
FAILED tests/unit_tests/experiment_plans/test_pin_tip_centring.py::test_trigger_and_return_pin_tip_works_for_AD_pin_tip_detection - assert (100, 200) == (200, 100)
FAILED tests/unit_tests/experiment_plans/test_rotation_scan_plan.py::test_rotation_plan_smargon_doesnt_move_xyz_if_not_given_in_params - AssertionError: assert 0.47 == 0
```

--- a/src/hyperion/device_setup_plans/setup_panda.py
+++ b/src/hyperion/device_setup_plans/setup_panda.py
@@ -4,7 +4,7 @@ from pathlib import Path
 
 import bluesky.plan_stubs as bps
 from blueapi.core import MsgGenerator
-from dodal.beamlines.beamline_utils import get_directory_provider
+from dodal.common.beamlines.beamline_utils import get_directory_provider
 from dodal.devices.panda_fast_grid_scan import PandAGridScanParams
 from ophyd_async.core import load_device
 from ophyd_async.panda import (

--- a/src/hyperion/utils/aperturescatterguard.py
+++ b/src/hyperion/utils/aperturescatterguard.py
@@ -1,4 +1,4 @@
-from dodal.beamlines.beamline_parameters import get_beamline_parameters
+from dodal.common.beamlines.beamline_parameters import get_beamline_parameters
 from dodal.devices.aperturescatterguard import AperturePositions, ApertureScatterguard
 
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -139,8 +139,8 @@ def pytest_runtest_setup(item):
 
 
 def pytest_runtest_teardown(item):
-    if "dodal.beamlines.beamline_utils" in sys.modules:
-        sys.modules["dodal.beamlines.beamline_utils"].clear_devices()
+    if "dodal.common.beamlines.beamline_utils" in sys.modules:
+        sys.modules["dodal.common.beamlines.beamline_utils"].clear_devices()
     markers = [m.name for m in item.own_markers]
     if "skip_log_setup" in markers:
         _reset_loggers([*ALL_LOGGERS, dodal_logger])

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -12,8 +12,9 @@ import bluesky.plan_stubs as bps
 import pytest
 from bluesky.run_engine import RunEngine
 from bluesky.utils import Msg
-from dodal.beamlines import beamline_utils, i03
-from dodal.beamlines.beamline_parameters import (
+from dodal.beamlines import i03
+from dodal.common.beamlines import beamline_utils
+from dodal.common.beamlines.beamline_parameters import (
     GDABeamlineParameters,
 )
 from dodal.devices.aperturescatterguard import (

--- a/tests/system_tests/experiment_plans/test_fgs_plan.py
+++ b/tests/system_tests/experiment_plans/test_fgs_plan.py
@@ -8,7 +8,7 @@ import pytest
 import pytest_asyncio
 from bluesky.run_engine import RunEngine
 from dodal.beamlines import i03
-from dodal.beamlines.beamline_parameters import (
+from dodal.common.beamlines.beamline_parameters import (
     BEAMLINE_PARAMETER_PATHS,
     GDABeamlineParameters,
 )

--- a/tests/system_tests/hyperion/test_aperturescatterguard_system.py
+++ b/tests/system_tests/hyperion/test_aperturescatterguard_system.py
@@ -1,5 +1,5 @@
 import pytest
-from dodal.beamlines.beamline_parameters import (
+from dodal.common.beamlines.beamline_parameters import (
     BEAMLINE_PARAMETER_PATHS,
     GDABeamlineParameters,
 )

--- a/tests/unit_tests/device_setup_plans/test_dcm_pitch_roll_mirror_adjuster.py
+++ b/tests/unit_tests/device_setup_plans/test_dcm_pitch_roll_mirror_adjuster.py
@@ -3,7 +3,7 @@ from unittest.mock import MagicMock, PropertyMock, patch
 
 import pytest
 from bluesky.run_engine import RunEngine
-from dodal.beamlines.beamline_parameters import GDABeamlineParameters
+from dodal.common.beamlines.beamline_parameters import GDABeamlineParameters
 from dodal.devices.focusing_mirror import (
     FocusingMirrorWithStripes,
     MirrorStripe,

--- a/tests/unit_tests/experiment_plans/test_flyscan_xray_centre_plan.py
+++ b/tests/unit_tests/experiment_plans/test_flyscan_xray_centre_plan.py
@@ -10,7 +10,7 @@ import pytest
 from bluesky.run_engine import RunEngine
 from bluesky.utils import FailedStatus, Msg
 from dodal.beamlines import i03
-from dodal.beamlines.beamline_utils import clear_device
+from dodal.common.beamlines.beamline_utils import clear_device
 from dodal.devices.detector.det_dim_constants import (
     EIGER2_X_4M_DIMENSION,
     EIGER_TYPE_EIGER2_X_4M,

--- a/tests/unit_tests/experiment_plans/test_grid_detection_plan.py
+++ b/tests/unit_tests/experiment_plans/test_grid_detection_plan.py
@@ -80,7 +80,10 @@ def fake_devices(RE, smargon: Smargon, backlight: Backlight, test_config_files):
         yield composite, mock_image
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 def test_grid_detection_plan_runs_and_triggers_snapshots(
     RE: RunEngine,
@@ -104,7 +107,10 @@ def test_grid_detection_plan_runs_and_triggers_snapshots(
     assert image.save.call_count == 6
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 @pytest.mark.asyncio
 async def test_grid_detection_plan_gives_warningerror_if_tip_not_found(
@@ -138,7 +144,10 @@ async def test_grid_detection_plan_gives_warningerror_if_tip_not_found(
     assert "No pin found" in excinfo.value.args[0]
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 def test_given_when_grid_detect_then_start_position_as_expected(
     fake_devices,
@@ -182,7 +191,10 @@ def test_given_when_grid_detect_then_start_position_as_expected(
     assert gridscan_params["z_start_um"] == pytest.approx(-0.0001)
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 @patch(
     "hyperion.experiment_plans.oav_grid_detection_plan.pre_centring_setup_oav",
@@ -212,7 +224,10 @@ def test_when_grid_detection_plan_run_twice_then_values_do_not_persist_in_callba
         RE(decorated())
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 def test_when_grid_detection_plan_run_then_ispyb_callback_gets_correct_values(
     fake_devices, RE: RunEngine, test_config_files, test_fgs_params
@@ -272,7 +287,10 @@ def test_when_grid_detection_plan_run_then_ispyb_callback_gets_correct_values(
         )
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 def test_when_grid_detection_plan_run_then_grid_detection_callback_gets_correct_values(
     fake_devices, RE: RunEngine, test_config_files, test_fgs_params
@@ -318,7 +336,10 @@ def test_when_grid_detection_plan_run_then_grid_detection_callback_gets_correct_
     "odd",
     [(True), (False)],
 )
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("bluesky.plan_stubs.sleep", new=MagicMock())
 @patch("hyperion.experiment_plans.oav_grid_detection_plan.LOGGER")
 def test_when_detected_grid_has_odd_y_steps_then_add_a_y_step_and_shift_grid(

--- a/tests/unit_tests/experiment_plans/test_optimise_attenuation_plan.py
+++ b/tests/unit_tests/experiment_plans/test_optimise_attenuation_plan.py
@@ -35,7 +35,8 @@ def mock_emit():
     LOGGER.removeHandler(test_handler)
 
 
-def fake_create_devices() -> OptimizeAttenuationComposite:
+@pytest.fixture
+def fake_composite() -> OptimizeAttenuationComposite:
     sample_shutter = i03.sample_shutter(
         fake_with_ophyd_sim=True, wait_for_connection=True
     )
@@ -51,6 +52,20 @@ def get_good_status():
     status = Status()
     status.set_finished()
     return status
+
+
+@pytest.fixture
+def fake_composite_mocked_sets(fake_composite: OptimizeAttenuationComposite):
+    with patch.object(
+        fake_composite.xspress3mini.do_arm,
+        "set",
+        MagicMock(return_value=get_good_status()),
+    ), patch.object(
+        fake_composite.sample_shutter,
+        "set",
+        MagicMock(return_value=get_good_status()),
+    ):
+        yield fake_composite
 
 
 def test_is_deadtime_optimised_returns_true_once_direction_is_flipped_and_deadtime_goes_back_above_threshold(
@@ -76,15 +91,14 @@ def test_is_deadtime_is_optimised_logs_warning_when_upper_transmission_limit_is_
 
 
 def test_total_counts_calc_new_transmission_raises_warning_on_high_transmission(
-    RE: RunEngine, mock_emit
+    RE: RunEngine, mock_emit, fake_composite_mocked_sets: OptimizeAttenuationComposite
 ):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-    composite.sample_shutter.set = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.do_arm.set = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.dt_corrected_latest_mca.sim_put([1, 1, 1, 1, 1, 1])  # type: ignore
+    fake_composite_mocked_sets.xspress3mini.dt_corrected_latest_mca.sim_put(  # type: ignore
+        [1, 1, 1, 1, 1, 1]
+    )
     RE(
         total_counts_optimisation(
-            composite,
+            fake_composite_mocked_sets,
             transmission=0.1,
             low_roi=0,
             high_roi=1,
@@ -122,13 +136,12 @@ def test_calculate_new_direction_gives_correct_value(
     autospec=True,
 )
 def test_deadtime_optimisation_calculates_deadtime_correctly(
-    mock_do_device_optimise_iteration, RE: RunEngine
+    mock_do_device_optimise_iteration,
+    RE: RunEngine,
+    fake_composite: OptimizeAttenuationComposite,
 ):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-
-    composite.xspress3mini.channel_1.total_time.sim_put(100)  # type: ignore
-    composite.xspress3mini.channel_1.reset_ticks.sim_put(101)  # type: ignore
-    is_deadtime_optimised.return_value = True
+    fake_composite.xspress3mini.channel_1.total_time.sim_put(100)  # type: ignore
+    fake_composite.xspress3mini.channel_1.reset_ticks.sim_put(101)  # type: ignore
 
     with patch(
         "hyperion.experiment_plans.optimise_attenuation_plan.is_deadtime_optimised",
@@ -136,7 +149,7 @@ def test_deadtime_optimisation_calculates_deadtime_correctly(
     ) as mock_is_deadtime_optimised:
         RE(
             deadtime_optimisation(
-                composite,
+                fake_composite,
                 0.5,
                 2,
                 0.01,
@@ -203,22 +216,28 @@ def test_is_counts_within_target_is_false(total_count, lower_limit, upper_limit)
     assert is_counts_within_target(total_count, lower_limit, upper_limit) is False
 
 
-def test_total_count_exception_raised_after_max_cycles_reached(RE: RunEngine):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-    composite.sample_shutter.set = MagicMock(return_value=get_good_status())
+def test_total_count_exception_raised_after_max_cycles_reached(
+    RE: RunEngine, fake_composite_mocked_sets: OptimizeAttenuationComposite
+):
     optimise_attenuation_plan.is_counts_within_target = MagicMock(return_value=False)
-    composite.xspress3mini.arm = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.dt_corrected_latest_mca.sim_put([1, 1, 1, 1, 1, 1])  # type: ignore
+    fake_composite_mocked_sets.xspress3mini.dt_corrected_latest_mca.sim_put(  # type: ignore
+        [1, 1, 1, 1, 1, 1]
+    )
     with pytest.raises(AttenuationOptimisationFailedException):
-        RE(total_counts_optimisation(composite, 1, 0, 10, 0, 5, 2, 1, 0, 0))
+        RE(
+            total_counts_optimisation(
+                fake_composite_mocked_sets, 1, 0, 10, 0, 5, 2, 1, 0, 0
+            )
+        )
 
 
-def test_arm_devices_runs_correct_functions(RE: RunEngine):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-    composite.xspress3mini.detector_state.sim_put("Acquire")  # type: ignore
-    composite.xspress3mini.arm = MagicMock(return_value=get_good_status())
-    RE(arm_devices(composite.xspress3mini))
-    composite.xspress3mini.arm.assert_called_once()
+def test_arm_devices_runs_correct_functions(
+    RE: RunEngine, fake_composite: OptimizeAttenuationComposite
+):
+    fake_composite.xspress3mini.detector_state.sim_put("Acquire")  # type: ignore
+    fake_composite.xspress3mini.arm = MagicMock(return_value=get_good_status())
+    RE(arm_devices(fake_composite.xspress3mini))
+    fake_composite.xspress3mini.arm.assert_called_once()
 
 
 @pytest.mark.parametrize(
@@ -246,16 +265,15 @@ def test_deadtime_calc_new_transmission_raises_error_on_low_ransmission():
 
 
 def test_total_count_calc_new_transmission_raises_error_on_low_ransmission(
-    RE: RunEngine,
+    RE: RunEngine, fake_composite_mocked_sets: OptimizeAttenuationComposite
 ):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-    composite.xspress3mini.dt_corrected_latest_mca.sim_put([1, 1, 1, 1, 1, 1])  # type: ignore
-    composite.sample_shutter.set = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.do_arm.set = MagicMock(return_value=get_good_status())
+    fake_composite_mocked_sets.xspress3mini.dt_corrected_latest_mca.sim_put(  # type: ignore
+        [1, 1, 1, 1, 1, 1]
+    )
     with pytest.raises(AttenuationOptimisationFailedException):
         RE(
             total_counts_optimisation(
-                composite,
+                fake_composite_mocked_sets,
                 1e-6,
                 0,
                 1,
@@ -270,26 +288,26 @@ def test_total_count_calc_new_transmission_raises_error_on_low_ransmission(
 
 
 @patch("hyperion.experiment_plans.optimise_attenuation_plan.arm_devices", autospec=True)
-def test_total_counts_gets_within_target(mock_arm_devices, RE: RunEngine):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-
+def test_total_counts_gets_within_target(
+    mock_arm_devices,
+    RE: RunEngine,
+    fake_composite_mocked_sets: OptimizeAttenuationComposite,
+):
     # For simplicity we just increase the data array each iteration. In reality it's the transmission value that affects the array
     def update_data(_):
         nonlocal iteration
         iteration += 1
-        composite.xspress3mini.dt_corrected_latest_mca.sim_put(  # type: ignore
+        fake_composite_mocked_sets.xspress3mini.dt_corrected_latest_mca.sim_put(  # type: ignore
             ([50, 50, 50, 50, 50]) * iteration
         )
         return get_good_status()
 
-    composite.attenuator.set = update_data
+    fake_composite_mocked_sets.attenuator.set = update_data
     iteration = 0
-    composite.sample_shutter.set = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.do_arm.set = MagicMock(return_value=get_good_status())
 
     RE(
         total_counts_optimisation(
-            composite,
+            fake_composite_mocked_sets,
             transmission=1,
             low_roi=0,
             high_roi=4,
@@ -325,14 +343,16 @@ def test_optimisation_attenuation_plan_runs_correct_functions(
     mock_total_counts_optimisation,
     optimisation_type,
     RE: RunEngine,
+    fake_composite: OptimizeAttenuationComposite,
 ):
-    composite: OptimizeAttenuationComposite = fake_create_devices()
-    composite.attenuator.set = MagicMock(return_value=get_good_status())
-    composite.xspress3mini.acquire_time.set = MagicMock(return_value=get_good_status())
+    fake_composite.attenuator.set = MagicMock(return_value=get_good_status())
+    fake_composite.xspress3mini.acquire_time.set = MagicMock(
+        return_value=get_good_status()
+    )
 
     RE(
         optimise_attenuation_plan.optimise_attenuation_plan(
-            composite,
+            fake_composite,
             optimisation_type=optimisation_type,
         )
     )
@@ -343,6 +363,6 @@ def test_optimisation_attenuation_plan_runs_correct_functions(
     else:
         mock_total_counts_optimisation.assert_not_called()
         mock_deadtime_optimisation.assert_called_once()
-    composite.attenuator.set.assert_called_once()
+    fake_composite.attenuator.set.assert_called_once()
     mock_check_parameters.assert_called_once()
-    composite.xspress3mini.acquire_time.set.assert_called_once()
+    fake_composite.xspress3mini.acquire_time.set.assert_called_once()

--- a/tests/unit_tests/experiment_plans/test_rotation_scan_plan.py
+++ b/tests/unit_tests/experiment_plans/test_rotation_scan_plan.py
@@ -150,7 +150,10 @@ def test_rotation_scan_calculations(test_rotation_params: RotationScan):
     assert motion_values.distance_to_move_deg == -180.3075
 
 
-@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch(
+    "dodal.common.beamlines.beamline_utils.active_device_is_same_type",
+    lambda a, b: True,
+)
 @patch("hyperion.experiment_plans.rotation_scan_plan.rotation_scan_plan", autospec=True)
 def test_rotation_scan(
     plan: MagicMock,

